pinot-minion/src/main/java/org/apache/pinot/minion/MinionConf.java

```diff
@@ -58,6 +59,9 @@ public int getPort() {
   }
 
   public String getInstanceId() {
+    if (StringUtils.isNotBlank(getProperty(CommonConstants.Minion.CONFIG_OF_MINION_ID))) {
+      return getProperty(CommonConstants.Minion.CONFIG_OF_MINION_ID);
+    }
```

Comment:
(minor) maybe keeping the behavior consistent, and do null check for all of them
```suggestion
    String instanceId = getProperty(CommonConstants.Minion.CONFIG_OF_MINION_ID);
    return instanceId != null ? instanceId : getProperty(CommonConstants.Helix.Instance.INSTANCE_ID_KEY);
```